pkg/installmanager/awsactuator.go

```diff
@@ -0,0 +1,113 @@
+package installmanager
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/s3"
+	log "github.com/sirupsen/logrus"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	hivev1 "github.com/openshift/hive/pkg/apis/hive/v1"
+	awsclient "github.com/openshift/hive/pkg/awsclient"
+	"github.com/openshift/hive/pkg/constants"
+
+	restful "github.com/emicklei/go-restful"
+	"github.com/pkg/errors"
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+)
+
+func init() {
+	registerActuator(&awsActuator{awsClientFn: getAWSClient})
+}
+
+// Ensure AWSActuator implements the Actuator interface. This will fail at compile time when false.
+var _ Actuator = &awsActuator{}
+
+// AWSActuator manages getting the desired state, getting the current state and reconciling the two.
+type awsActuator struct {
+	// awsClientFn is the function to build an AWS client, here for testing
+	awsClientFn func(client.Client, string, string, string, log.FieldLogger) (awsclient.Client, error)
+}
+
+// CanHandle returns true if the actuator can handle a particular ClusterDeprovision
+func (a *awsActuator) CanHandle() bool {
+	provider, foundProviderEnvVar := os.LookupEnv(constants.InstallLogsUploadProviderEnvVar)
+	if !foundProviderEnvVar {
+		log.Debug("Couldn't find install logs provider environment variable. Skipping.")
+		return false
+	}
+
+	return provider == constants.InstallLogsUploadProviderAWS
+}
+
+// UploadLogs uploads installer logs to the provider's storage mechanism.
+func (a *awsActuator) UploadLogs(clusterName string, clusterprovision *hivev1.ClusterProvision, c client.Client, log log.FieldLogger, filenames ...string) error {
+	secretName, foundSecretName := os.LookupEnv(constants.InstallLogsCredentialsSecretRefEnvVar)
+	if !foundSecretName {
+		return errors.New("couldn't find secret name in environment variable. Skipping upload")
+	}
```

Comment:
I would move all these checks into CanHandle/IsConfigured so we can differentiate between are we configured, vs an error uploading a little more clearly.